crates/circuit/src/operations.rs

```diff
@@ -2396,6 +2426,132 @@ impl Operation for StandardGate {
+                _ => None,
+            },
+            Self::X => match params {
+                [] => Some(gate_matrix::X_GATE),
+                _ => None,
+            },
+            Self::Y => match params {
+                [] => Some(gate_matrix::Y_GATE),
+                _ => None,
+            },
+            Self::Z => match params {
+                [] => Some(gate_matrix::Z_GATE),
+                _ => None,
+            },
+            Self::Phase => match params {
+                [Param::Float(theta)] => Some(gate_matrix::phase_gate(*theta)),
+                _ => None,
+            },
+            Self::R => match params {
+                [Param::Float(theta), Param::Float(phi)] => Some(gate_matrix::r_gate(*theta, *phi)),
+                _ => None,
+            },
+            Self::RX => match params {
+                [Param::Float(theta)] => Some(gate_matrix::rx_gate(*theta)),
+                _ => None,
+            },
+            Self::RY => match params {
+                [Param::Float(theta)] => Some(gate_matrix::ry_gate(*theta)),
+                _ => None,
+            },
+            Self::RZ => match params {
+                [Param::Float(theta)] => Some(gate_matrix::rz_gate(*theta)),
+                _ => None,
+            },
+            Self::S => match params {
+                [] => Some(gate_matrix::S_GATE),
+                _ => None,
+            },
+            Self::Sdg => match params {
+                [] => Some(gate_matrix::SDG_GATE),
+                _ => None,
+            },
+            Self::SX => match params {
+                [] => Some(gate_matrix::SX_GATE),
+                _ => None,
+            },
+            Self::SXdg => match params {
+                [] => Some(gate_matrix::SXDG_GATE),
+                _ => None,
+            },
+            Self::T => match params {
+                [] => Some(gate_matrix::T_GATE),
+                _ => None,
+            },
+            Self::Tdg => match params {
+                [] => Some(gate_matrix::TDG_GATE),
+                _ => None,
+            },
+            Self::U => match params {
+                [Param::Float(theta), Param::Float(phi), Param::Float(lam)] => {
+                    Some(gate_matrix::u_gate(*theta, *phi, *lam))
+                }
+                _ => None,
+            },
+            Self::U1 => match params[0] {
+                Param::Float(val) => Some(gate_matrix::u1_gate(val)),
+                _ => None,
+            },
+            Self::U2 => match params {
+                [Param::Float(phi), Param::Float(lam)] => Some(gate_matrix::u2_gate(*phi, *lam)),
+                _ => None,
+            },
+            Self::U3 => match params {
+                [Param::Float(theta), Param::Float(phi), Param::Float(lam)] => {
+                    Some(gate_matrix::u3_gate(*theta, *phi, *lam))
+                }
+                _ => None,
+            },
+            Self::CH => None,
```

Comment:
Instead of typing all of them out maybe just `_ => None`? 🙂 (incl. the GlobalPhaseGate)

Reply:
I'm generally in favour of not using the wildcard match - we've already had a few bugs because of wildcard matching as "generically not handled" and then _whoops_ the data model expands and we forgot to update a bunch of places.

Reply:
Yeah I agree in general, but here I thought we won't add standard gates anyways and it would save a lot of code 😄 but I'm also fine always expanding if the wildcard match passes silently 🙂

Reply:
I am fine either way, with slightly leaning on the side of explicitly mentioning all gates, for the reason that Jake mentioned. And you never know, we may want to add another standard gate in the future.